Kudu.FunctionalTests/Vfs/VfsControllerBaseTest.cs

```diff
@@ -393,7 +393,10 @@ public VfsControllerBaseTest(RemoteVfsManager client, bool testConflictingUpdate
                 IEnumerable<HttpResponseMessage> concurrentResponses = concurrentUpdates.Select(update => update.Result);
                 foreach (HttpResponseMessage concurrentResponse in concurrentResponses)
                 {
-                    Assert.Equal(HttpStatusCode.NoContent, concurrentResponse.StatusCode);
+                    Assert.True(
+                       concurrentResponse.StatusCode == HttpStatusCode.NoContent ||
+                       concurrentResponse.StatusCode == HttpStatusCode.PreconditionFailed,
+                       $"Status code expected to be either {HttpStatusCode.NoContent} or {HttpStatusCode.PreconditionFailed} but got {concurrentResponse.StatusCode}");
```

Comment:
I'd just add a comment line that explains why getting PreconditionFailed is sometimes expected.